packages/shared/src/components/modals/common/types.ts

```diff
@@ -0,0 +1,26 @@
+import { KeyboardEvent, MouseEvent, createContext } from 'react';
+
+export enum ModalKind {
+  FlexibleCenter = 'flexible-center',
+  FlexibleTop = 'flexible-top',
+  FixedCenter = 'fixed-center',
+}
+
+export enum ModalSize {
+  XSmall = 'x-small',
+  Small = 'small',
+  Medium = 'medium',
+  Large = 'large',
+}
+
+export type ModalContextProps = {
+  onRequestClose: null | ((event: MouseEvent | KeyboardEvent) => void);
+  kind: ModalKind;
+  size: ModalSize;
+};
+
+export const ModalPropsContext = createContext<ModalContextProps>({
+  onRequestClose: null,
+  kind: ModalKind.FlexibleCenter,
+  size: ModalSize.Medium,
+});
```

Comment:
Why do we favor context here instead of passing directly to the modal?

Reply:
We do pass it directly to the modal, the context is used inside the modal

Reply:
Happy to jump on a call if it is not clear